core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -749,27 +783,64 @@ protected ComputerLauncher grabLauncher(Node node) {
     }
 
     /**
-     * Get the agent version
+     * Get the agent version.
+     * @return Agent version if the agent is online and has a modern remoting version (2.0+).
+     *         Otherwise it may return a diagnostics information.
      */
+    @Nonnull
     public String getSlaveVersion() throws IOException, InterruptedException {
+        if (channel == null) {
+            return "Unknown (agent is offline)";
+        }
         return channel.call(new SlaveVersion());
     }
+    
+    //TODO: move to public API and deprecate the old one after the merge to master
```

Comment:
I want to switch to the `CheckForNull` API, and it is a good opportunity to get rid of the "Slave" in API definition. Can remove it from this commit if you prefer